### ISSUE TITLE
Add `IAuthenticationConfigurationProvider.GetAuthenticationConfiguration`

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#41989]
```diff
+ Microsoft.AspNetCore.Authentication.IAuthenticationConfigurationProvider.GetAuthenticationConfiguration() -> Microsoft.Extensions.Configuration.IConfiguration!
```

```diff
+ Microsoft.AspNetCore.Authentication.AuthenticationConfigurationConstants
+ const Microsoft.AspNetCore.Authentication.AuthenticationConfigurationConstants.Authentication = "Authentication" -> string!
+ const Microsoft.AspNetCore.Authentication.AuthenticationConfigurationConstants.DefaultScheme = "DefaultScheme" -> string!
+ const Microsoft.AspNetCore.Authentication.AuthenticationConfigurationConstants.Schemes = "Schemes" -> string!
```

**Comment by davidfowl** [comment]
Why does this need the schemes as a first class property?

**Comment by captainsafia** [comment]
> Why does this need the schemes as a first class property?

It makes sense to me to have every key we support under Authentication as a first-class property. I suppose on alternative is to store an `Authentication:Schemes` key but it made sense to keep the properties separate and leave it up to the implementor to combine as needed.

**Comment by davidfowl** [comment]
> It makes sense to me to have every key we support under Authentication as a first-class property. I suppose on alternative is to store an Authentication:Schemes key but it made sense to keep the properties separate and leave it up to the implementor to combine as needed.

Do we need a strongly typed object model on authentication configuration for every top level configuration property? I think we should keep this type simple. It's a way to get access to the top level auth config section, and the config section for a scheme. Otherwise, it's no longer an interface but instead a poco like AuthenticationOptions.

GetAuthenticationConfiguration should be a property.

**Comment by halter73** [comment]
Here's the proposed diff from the originally approved API if I'm reading this correctly:

```diff
namespace Microsoft.AspNetCore.Authentication;

public interface IAuthenticationConfigurationProvider
{
    public IConfiguration Authentication { get; }
-    public IConfiguration AuthenticationSchemes { get; }
}

+ public static class AuthenticationConfigurationProviderExtensions
+ {
+    public static IConfiguration GetSchemeConfiguration(this IAuthenticationConfigurationProvider provider, string authenticationScheme);
+ }
```

Is there a benefit of calling `_authenticationConfigurationProvider.GetSchemeConfiguration(name);` over `_authenticationConfigurationProvider.GetSection($"Scheme:{name}")`? Is it just trying to get rid of the magic string? Seems reasonable

It does seem a little weird to call  `_authenticationConfigurationProvider.Authentication` for the parent config section and call `_authenticationConfigurationProvider.GetSchemeConfiguration(name)` for the subsection because one calls out `Configuration` in the name and the other doesn't. Should  `_authenticationConfigurationProvider.Authentication` be `_authenticationConfigurationProvider.AuthenticationConfiguration` instead?

**Comment by captainsafia** [comment]
> Is there a benefit of calling _authenticationConfigurationProvider.GetSchemeConfiguration(name); over _authenticationConfigurationProvider.GetSection($"Scheme:{name}")? Is it just trying to get rid of the magic string? Seems reasonable

Yep, and to provide an abstraction over getting the scheme from the provider, which we will be using in all the different options from config implementations (e.g. `JwtBearerConfigureOptions`, `CookieConfigureOptions`)

> Should _authenticationConfigurationProvider.Authentication be _authenticationConfigurationProvider.AuthenticationConfiguration instead?

Sure. That works.

**Comment by halter73** [comment]
Approved API (relative to what's currently checked in):

```diff
namespace Microsoft.AspNetCore.Authentication;

public interface IAuthenticationConfigurationProvider
{
-    IConfiguration GetAuthenticationSchemeConfiguration(string authenticationScheme);
+    IConfiguration AuthenticationConfiguration { get; }
}

+ public static class AuthenticationConfigurationProviderExtensions
+ {
+    public static IConfiguration GetSchemeConfiguration(this IAuthenticationConfigurationProvider provider, string authenticationScheme);
+ }
```
